java/src/main/java/ai/rapids/cudf/CudaComputeMode.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.rapids.cudf;
+
+/**
+ * This is the Java mapping of CUDA device compute modes.
+ */
+public enum CudaComputeMode {
+  /**
+   * Default compute mode
+   * Multiple threads can use cudaSetDevice() with this device.
+   */
+  DEFAULT(0),
+  /**
+   * Compute-exclusive-thread mode
+   * Only one thread in one process will be able to use cudaSetDevice() with this device.
+   */
```

Comment:
This mode is deprecated. Better to add the same comment here.
```Bash
(base) liangcail@liangcail-ubuntu18:~/work/projects/on_github/spark-rapids$ nvidia-smi -c 1
Warning: Exclusive_Thread was deprecated! Setting Exclusive_Process instead.
Unable to set the compute mode for GPU 00000000:01:00.0: Insufficient Permissions
Terminating early due to previous errors.

```

Reply:
Added warning message for the deprecation.